### PR TITLE
Replace last uses of MSVC-specific `_DEBUG` with `HS_DEBUGGING`

### PR DESCRIPTION
**Pull request by dgelessus**
This allows removing some compiler-specific CMake code.

### CODE REVIEW

**Comment by dgelessus** [comment]
Clearly I didn't grep right the first time and missed a few more uses of `_DEBUG`, sorry about that.

But if I'm reading the Linux CI failures correctly, the PhysX headers expect `_DEBUG` to be defined in debug mode, even for non-MSVC compilers? In that case, the extra CMake code probably can't be removed after all.

**Comment by Hoikas** [comment]
We should probably move the definition into the PhysX target in that case.

**Comment by dgelessus** [comment]
Should both be fixed now. The `PhysX` library targets now define `_DEBUG` as needed. All uses of `_DEBUG` are gone, except for one in Sources/Tools/MaxPlasmaLights/plRTLights.rc, because `HS_DEBUGGING` isn't available there and that code is Windows-specific anyway.

**Review by dpogue — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,10 @@ option(PRODUCT_EMBED_BUILD_INFO "Embed build revision information into plProduct
 cmake_dependent_option(PRODUCT_EMBED_BUILD_TIME "Embed build time into plProduct" ON [[PRODUCT_EMBED_BUILD_INFO]] OFF)
 
 # HeadSpin Configuration
+
+# Define HS_DEBUGGING for debug builds
+add_compile_definitions($<$<CONFIG:Debug>:HS_DEBUGGING>)
```
Comment (Hoikas) [inline CMakeLists.txt:39]: Ideally we would be moving away from these scoped commands and toward target based stuff, but there's already a precedent for this below, so I don't think this is a blocker.

**Review by Hoikas — APPROVED** [review]

**Pull request merged**